src/QLD.cpp

```diff
@@ -43,15 +43,23 @@ int QLD::fail() const
   return fail_;
 }
 
-void QLD::problem(int nrvar, int nreq, int nrineq, int ldq)
+void QLD::problem(int nrvar, int nreq, int nrineq, int ldq, int lda)
+{
+  problemNoOverhead(nrvar, nreq, nrineq, ldq, lda);
+
+  int nrconstr = nreq + nrineq;
+  A_.resize(nrconstr, nrvar);
+  B_.resize(nrconstr);
+}
+
+void QLD::problemNoOverhead(int nrvar, int nreq, int nrineq, int ldq, int lda)
 {
   int nrconstr = nreq + nrineq;
 
   if(ldq < nrvar) ldq = nrvar;
+  if(lda < nrconstr) lda = nrconstr;
 
-  assert(ldq >= nrvar);
-
-  int MMAX = nrconstr == 0 ? 1 : nrconstr;
+  int MMAX = lda == 0 ? 1 : lda;
   int NMAX = ldq == 0 ? 1 : ldq;
 
   A_.resize(nrconstr, nrvar);
```

Comment:
I think you meant to remove this part from `::problemNoOverhead`. Otherwise it doesn't make much sense to split the two functions